### PR TITLE
simplifyTestKit doc async sample #18804

### PR DESCRIPTION
**Pull request by richardimaoka**
I tested this locally in my desktop, but assume the change will be auto-tested too?
Thanks

### CODE REVIEW

**Comment by johanandren** [comment]
OK TO TEST

File: akka-docs/rst/scala/code/docs/testkit/PlainWordSpec.scala [hunk 1]
```diff
@@ -13,12 +13,10 @@ import org.scalatest.Matchers
 import org.scalatest.BeforeAndAfterAll
 
 //#implicit-sender
-class MySpec(_system: ActorSystem) extends TestKit(_system) with ImplicitSender
+class MySpec() extends TestKit(ActorSystem("MySpec")) with ImplicitSender
```
Comment (johanandren) [inline akka-docs/rst/scala/code/docs/testkit/PlainWordSpec.scala]: Seems reasonable to me, as the sample does not show how to create an abstract base class for your tests, but an actual concrete test class

**Comment by johanandren** [comment]
Fixes #18804

**Comment by johanandren** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM, the `def this` has confused newcomers in the past indeed. 
Thanks for the contrib!

**Pull request merged**